controller/src/main/java/org/jboss/as/controller/xml/VersionedURN.java

```diff
@@ -51,12 +62,30 @@ public VersionedURN(String identifier, IntVersion version) {
         this(List.of(identifier), version);
     }
 
+    /**
+     * Constructs a versioned URN-based namespace using the specified namespace identifier and version.
+     * @param nid a namespace identifier
+     * @param version a version
+     */
+    public VersionedURN(String identifier, IntVersion version, Stream stream) {
+        this(List.of(identifier), version, stream);
+    }
+
     /**
      * Constructs a versioned URN-based namespace using the specified URN components and version.
      * @param components a list of URN components
      * @param version a version
      */
     public VersionedURN(List<String> identifiers, IntVersion version) {
-        super(String.join(URN_DELIMITER, new CompositeIterable<>(List.of(URN), identifiers, List.of(version.toString(2)))), version);
+        this(identifiers, version, Stream.DEFAULT);
+    }
+
+    /**
+     * Constructs a versioned URN-based namespace using the specified URN components and version.
+     * @param components a list of URN components
+     * @param version a version
+     */
+    public VersionedURN(List<String> identifiers, IntVersion version, Stream stream) {
+        super(String.join(URN_DELIMITER, new CompositeIterable<>(List.of(URN), identifiers, (stream == Stream.DEFAULT) ? List.of(version.toString(2)) : List.of(stream.toString(), version.toString(2)))), version);
```

Comment:
I feel like it should also the stable stream should also be a valid URN (ex. `urn:wildfly:elytron:17.0:stable`). Default representation still drops that component, but it should be valid.

Reply:
So, a given subsystem (or deployment descriptor) would need to register 2 namespaces with the same parser for the current stable version?  That seems overly cumbersome to me.

Reply:
Yeah, I'm not sure how I feel about that one, just seemed a bit inconsistent

Reply:
Keep in mind that most subsystems will not have any preview or experimental features (at least initially), and thus will not register any preview/experimental namespaces.
It seems odd to me that we would start registering duplicate namespaces for the stable stream only when they eventually incorporate a preview or experimental feature.

Reply:
Although I don't see a reason for a separate "default" URN to exist, that's just what the unspecified one means.

Reply:
Ultimately, every valid URN needs to be registered with a subsystem parser.  It is unclear to me when subsystem would need to register both a urn:wildfly:foo:X.Y and urn:wildfly:foo:stable:X.Y namespace.  Currently, subsystems only register namespaces using the former pattern.  None use the latter.  Do you propose we create these retroactively for every existing namespace for every subsystem?  What about external subsystems? e.g. those in wildfly-extras?

Reply:
Good point, leave it as is then :)